lntest/node.go

```diff
@@ -266,12 +271,50 @@ func (hn *HarnessNode) start(lndError chan<- error) error {
 	var errb bytes.Buffer
 	hn.cmd.Stderr = &errb
 
+	// Make sure the log file cleanup function is initialized, even
+	// if no log file is created.
+	hn.finalizeLogfile = func() {
+		if hn.logFile != nil {
+			hn.logFile.Close()
+		}
+	}
+
 	// If the logoutput flag is passed, redirect output from the nodes to
 	// log files.
 	if *logOutput {
 		fileName := fmt.Sprintf("output-%d-%s-%s.log", hn.NodeID,
 			hn.cfg.Name, hex.EncodeToString(hn.PubKey[:logPubKeyBytes]))
 
+		// If the node's PubKey is not yet initialized, create a temporary
+		// file name with a random part (so other nodes that would otherwise
+		// collide with the same file name don't accidentally write to the
+		// same file). Later, after the PubKey has been initialized, the
+		// file can be moved to its final name.
+		if bytes.Equal(hn.PubKey[:4], []byte{0, 0, 0, 0}) {
+			rand.Seed(time.Now().UnixNano())
+			randomBytes := make([]byte, 4)
+			rand.Read(randomBytes)
+			fileName = fmt.Sprintf("output-%d-%s-tmp__%s.log", hn.NodeID,
+				hn.cfg.Name, hex.EncodeToString(randomBytes))
+
+			// Once the node has done its work, the log file can be renamed.
+			hn.finalizeLogfile = func() {
```

Comment:
since this method is used only inseide start, it shouldn't need to be part of the node struct.